### DISCUSSION TITLE
Dependabot: configurable minimum severity level

### DISCUSSION THREAD

**Discussion by xomgc3** [community/community#46288] [Code Security]
### Select Topic Area

Product Feedback

### Body

Dependabot alerts and PRs generate a lot of noise.  Please allow a minimum severity level configuration via the org & repo UI, and the `dependabot.yml` config file.

For instance, Dependabot should not alert or submit fixes for severities below High.

**Comment by erinhav** [comment]
Hi @xomgc3, PM for Dependabot Alerts here. We're working on configuration now. 

For reporting purposes, what do you want to happen to alerts which are ignored? 

For example, if we're ignoring low-severity alerts:
- Automatically dismiss them as a `dismissed` alert with a special dismiss reason (of course, no notifications or PRs)
- Automatically close them as a special type of closed alert (e.g. "ignored") again with suppressed notifications and PRs

**Reply by joe-kimmel-vmw** [reply]
I have a related use-case that I'd like to comment here on, since this issue would partially service it - 

I'd really love to have patches of only Critical severity issues on old release branches ("as a maintainer, i want to define Long Term Support releases, and automate patching as possible to minimize the redundant effort of maintaining multiple LTS releases from backporting the same patches/bumps to multiple release branches")

so step 1 of that would be to be able to configure security patches to multiple branches simultaneously (ideally via wildcard, like release/*, but even doing it via hardcoded name would be fine since we're thinking about a few LTS releases per year),  and step 2 would be to be able to set the severity level,  so that e.g. my development trunk could get Medium+ issues patched, and the LTS releases could get only Critical (or only High + Critical,  depending on org policy...). 

does that make sense?  So for this feature to support the above, it's less important _how_ they're ignored, and more important that it's configured in a way that's flexible / future-forward w.r.t. automating the patching of multiple branches at once. 

anyways hopefully that's not too much of a tangent :)

**Reply by xomgc3** [reply]
👋🏻 Hello, @erinhav.  I would think that dismissing alerts below the threshold would be most appropriate.  In the case where Medium alerts fall under the threshold, we might only want to dismiss them temporarily.  However, Lows might need to be closed and fully "ignored".

**Reply by erinhav** [reply]
Thanks both! We're currently prototyping out the ability to ignore alerts. Baby steps, but hope to ship some improvements soon. 

What other kinds of criteria would you want to ignore alerts by? How important is org-level config for you?

**Reply by xomgc3** [reply]
Org-level is our primary need.  Repo-level is secondary.
